pyo3-macros-backend/src/pyfunction.rs

```diff
@@ -83,13 +86,157 @@ impl PyFunctionArgPyO3Attributes {
     }
 }
 
+type PyFunctionWarningMessageAttribute = KeywordAttribute<attributes::kw::message, LitStr>;
+type PyFunctionWarningCategoryAttribute = KeywordAttribute<attributes::kw::category, Path>;
+
+pub struct PyFunctionWarningAttribute {
+    pub message: PyFunctionWarningMessageAttribute,
+    pub category: Option<PyFunctionWarningCategoryAttribute>,
+    pub span: Span,
+}
+
+#[derive(PartialEq)]
+pub enum PyFunctionWarningCategory {
+    Path(Path),
+    UserWarning,
+    DeprecationWarning,
```

Comment:
I assume this one is unused for now. It's probably no big deal to leave it in, but a `todo` or `fixme` comment with the intend would be nice.